pontos/copyright/header.py

```diff
@@ -24,11 +24,74 @@
 import argparse
 import re
 
+from typing import Dict, Tuple, Union
 from datetime import datetime
 from glob import glob
 from pathlib import Path
 from subprocess import SubprocessError, check_output
 
+def _get_modified_year(f: Path) -> str:
+    return check_output(
+        f"cd {f.parents[0]} && git log -1 "
+        f"--format='%ad' --date='format:%Y' {f.name}",
+        shell=True,
+        universal_newlines=True,
+    ).rstrip()
+
+def _find_copyright(line: str, regex: re.Pattern) -> Tuple[bool, Union[Dict[str, Union[str, None]], None]]:
+    copyright_match = re.search(regex, line)
+    if copyright_match:
+        return (
+            True,
+            {
+                'creation_year': copyright_match.group(1),
+                'modification_year': copyright_match.group(2),
+                'vendor': copyright_match.group(3),
+            },
+        )
+    return False, None
+
+def _update_year(file: Path, new_modified_year: str) -> None:
+    # copyright year(s) regex
+    regex = re.compile(
+        "[Cc]opyright.*?(19[0-9]{2}|20[0-9]{2}) "
+        "?-? ?(19[0-9]{2}|20[0-9]{2})? (.reenbone*)"
+    )
+
+    with open(file, "r+") as fp:
```

Comment:
Doesn't this work too?
```suggestion
    with f.open("r+") as fp:
```